### PR TITLE
Update code to current LTS version

### PR DESCRIPTION
**Pull request by jacek-foremski**
> [EZP-27160](http://jira.ez.no/browse/EZP-27160)
> [EZP-27098](http://jira.ez.no/browse/EZP-27098)
#
My goal was to: 
- fix any errors that prevented the code from running in the last LTS
- fix unit tests and make them pass
- change the codebase to follow PSR-2 and PHP/Symfony best practices
- remove unnecessary code

### CODE REVIEW

File: Command/CreateTweetContentCommand.php [hunk 1]
```diff
@@ -25,9 +25,10 @@ protected function configure()
     protected function execute(InputInterface $input, OutputInterface $output)
     {
         $repository = $this->getContainer()->get('ezpublish.api.repository');
-        $repository->setCurrentUser(
-            $repository->getUserService()->loadUserByLogin('admin')
-        );
+        $userService = $repository->getUserService();
```
Comment (bdunogier) [inline Command/CreateTweetContentCommand.php]: Not related to this line, but there's a typo in this change's commit message: "Remove call to *depracated* method setCurrentUser".

File: eZ/Publish/FieldType/Tweet/Type.php [hunk 2]
```diff
@@ -206,20 +206,30 @@ public function validate(FieldDefinition $fieldDefinition, SPIValue $fieldValue)
...
-            if (!in_array($m[1], $validatorConfiguration['authorList'])) {
-                $errors[] = new ValidationError(
-                    "Twitter user %user% is not in the approved author list",
-                    null,
-                    array($m[1])
-                );
-            }
+        if (!$this->authorApproved($author, $validatorConfiguration)) {
+            $errors[] = new ValidationError(
+                "Twitter user %user% is not in the approved author list",
```
Comment (bdunogier) [inline eZ/Publish/FieldType/Tweet/Type.php]: Nitpick: single quotes :)

File: eZ/Publish/FieldType/Tweet/Type.php [hunk 3]
```diff
@@ -206,20 +206,30 @@ public function validate(FieldDefinition $fieldDefinition, SPIValue $fieldValue)
...
+                null,
+                array('%user%' => $m[1])
+            );
         }
 
         return $errors;
     }
+
+    private function authorApproved($author, $validatorConfiguration)
```
Comment (bdunogier) [inline eZ/Publish/FieldType/Tweet/Type.php]: Naming: i'd change this to `isAuthorApproved` (verb).

File: eZ/Publish/FieldType/Tweet/Type.php [hunk 4]
```diff
@@ -206,20 +206,30 @@ public function validate(FieldDefinition $fieldDefinition, SPIValue $fieldValue)
 
         // Tweet Url validation
         if (!preg_match('#^https?://twitter.com/([^/]+)/status/[0-9]+$#', $fieldValue->url, $m)) {
-            $errors[] = new ValidationError("Invalid twitter status url %url%", null, array($fieldValue->url));
+            $errors[] = new ValidationError(
+                "Invalid twitter status url %url%",
+                null,
+                array('%url%' => $fieldValue->url)
```
Comment (bdunogier) [inline eZ/Publish/FieldType/Tweet/Type.php]: Short array syntax while at it ?

File: eZ/Publish/FieldType/Tweet/Type.php [hunk 5]
```diff
@@ -211,6 +211,7 @@ public function validate(FieldDefinition $fieldDefinition, SPIValue $fieldValue)
                 null,
                 array('%url%' => $fieldValue->url)
             );
+            return $errors;
```
Comment (bdunogier) [inline eZ/Publish/FieldType/Tweet/Type.php]: CS: space before return.

File: Resources/config/fieldtypes.yml [hunk 6]
```diff
@@ -1,16 +1,12 @@
-parameters:
-    ezsystems.tweetbundle.fieldType.eztweet.class: EzSystems\TweetFieldTypeBundle\eZ\Publish\FieldType\Tweet\Type
-    ezsystems.tweetbundle.fieldType.eztweet.converter.class: EzSystems\TweetFieldTypeBundle\eZ\Publish\FieldType\Tweet\LegacyConverter
-
 services:
-    ezsystems.tweetbundle.fieldType.eztweet:
+    ezsystems.tweetbundle.fieldtype.eztweet:
         parent: ezpublish.fieldType
-        class: %ezsystems.tweetbundle.fieldType.eztweet.class%
+        class: 'EzSystems\TweetFieldTypeBundle\eZ\Publish\FieldType\Tweet\Type'
```
Comment (bdunogier) [inline Resources/config/fieldtypes.yml]: Unless I'm mistaken, the quotes aren't necessary (see the examples from http://symfony.com/doc/current/service_container.html). I'd remove them altogether.

**Review by bdunogier — CHANGES_REQUESTED** [review]
A couple minor remarks, but overall: good job !

**Comment by bdunogier** [comment]
Note that some of my comments show as outdated, as I have reviewed per commit. Don't forget to open each at least once before dismissing them :)

For the future, the CS change on the whole repository should maybe be done first, as it made it very hard to distinguish CS changes from code changes when reviewing the PR as a whole.

**Comment by jacek-foremski** [comment]
I made requested changes and few more :)

I agree with your comment about CS changes and I will do just that next time.

**Review by bdunogier — APPROVED** [review]

**Comment by bdunogier** [comment]
@natanael89 can you rebase on top of master ? A typo fix ("ocurred" / "occurred") was merged yesterday that conflicts with your CS changes.

Once that is done, we can merge it.

**Comment by jacek-foremski** [comment]
@bdunogier I rebased it, so it's ready to be merged.

File: DependencyInjection/EzSystemsTweetFieldTypeExtension.php [hunk 7]
```diff
@@ -16,19 +16,19 @@
  */
 class EzSystemsTweetFieldTypeExtension extends Extension implements PrependExtensionInterface
 {
-    public function prepend( ContainerBuilder $container )
+    public function prepend(ContainerBuilder $container)
     {
-        $config = Yaml::parse( __DIR__ . '/../Resources/config/ez_field_templates.yml' );   
-        $container->prependExtensionConfig( 'ezpublish', $config );
+        $config = Yaml::parse(__DIR__ . '/../Resources/config/ez_field_templates.yml');
```
Comment (lolautruche) [inline DependencyInjection/EzSystemsTweetFieldTypeExtension.php]: `Yaml::parse()` must take YAML content now:

```php
$config = Yaml::parse(file_get_contents(__DIR__ . '/../Resources/config/ez_field_templates.yml'));
```

File: Resources/views/fields/eztweet.html.twig [hunk 8]
```diff
@@ -1,10 +1,10 @@
...
-    {% endset %}
-    {{ block( 'simple_block_field' ) }}
-{% endspaceless %}
+    {% spaceless %}
+        {% set field_value %}
+            {{ field.value.contents|raw }}
+        {% endset %}
+        {{ block( 'simple_block_field' ) }}
+    {% endspaceless %}
 {% endblock %}
```
Comment (lolautruche) [inline Resources/views/fields/eztweet.html.twig]: Missing empty line at end of file

File: Resources/views/platformui/content_type/view/eztweet.html.twig [hunk 9]
```diff
@@ -0,0 +1,2 @@
+{% block eztweet_settings %}
+{% endblock %}
```
Comment (lolautruche) [inline Resources/views/platformui/content_type/view/eztweet.html.twig]: Missing empty line

**Review by lolautruche — APPROVED** [review]
Besides my comment about YAML loading and minor stuff, +1

**Comment by bdunogier** [comment]
We will merge once @lolautruche's suggestions have been integrated. Thank you !

**Comment by jacek-foremski** [comment]
Thank you for your comments @lolautruche! Your suggestions have been added.

**Review by andrerom — APPROVED** [review]

**Comment by mere-teresa** [comment]
Thank you @natanael89 !

**Pull request merged**